docs/contributing-to-airbyte/resources/code-formatting.md

```diff
@@ -4,12 +4,11 @@
 
 ### 🐍 Python
 
-We format our Python code using:
+We use [Ruff](https://docs.astral.sh) for Python code formatting and import sorting. Our Ruff configuration is in the [pyproject.toml](https://github.com/airbytehq/airbyte/blob/master/pyproject.toml) file.
 
-- [Black](https://github.com/psf/black) for code formatting
-- [isort](https://pycqa.github.io/isort/) for import sorting
+Ruff is monorepo-friendly and supports nested inherited configuration; each sub-project can optionally override Ruff lint and formatting settings in their own `pyproject.toml` files, as needed per project.
 
-Our configuration for both tools is in the [pyproject.toml](https://github.com/airbytehq/airbyte/blob/master/pyproject.toml) file.
+Ruff [auto-detects the proper package classification]((https://docs.astral.sh/ruff/faq/#how-does-ruff-determine-which-of-my-imports-are-first-party-third-party-etc)) so that "local", "first party" and "third party" imports are sorted and grouped correctly, even within subprojects of the monorepo.
```

Comment:
```suggestion
Ruff [auto-detects the proper package classification](https://docs.astral.sh/ruff/faq/#how-does-ruff-determine-which-of-my-imports-are-first-party-third-party-etc) so that "local", "first party" and "third party" imports are sorted and grouped correctly, even within subprojects of the monorepo.
```